src/main/java/DukeManager.java

```diff
@@ -0,0 +1,147 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+class DukeManager {
+    private static final int TODO = 4;
+    private static final int DEADLINE = 9;
+    private static final int BY = 3;
+    private static final int EVENT = 6;
+    private static final int FROM = 5;
+    private static final int TO = 3;
+    private final ArrayList<Task>taskList;
+
+    DukeManager(ArrayList<Task>tl) {
+        this.taskList = tl;
+    }
+
+    public void run() {
+        Scanner io = new Scanner(System.in);
+        String inputt = io.nextLine();
+        while (!inputt.equals("bye")) {
+            try {
+                boolean isValidInput = false;
+                if (inputt.equals("list")) {
+                    isValidInput = true;
+                    printTasklist();
+                } else if (inputt.contains("mark")) {
+                    //if mark / unmark is detected
+                    isValidInput = true;
+                    markTask(inputt, this.taskList);
+                } else if (inputt.contains("delete")) {
+                    isValidInput = true;
+                    deleteTask(inputt, this.taskList);
+                } else {
+                    //else add task to array; create task based on input by user
+                    if (inputt.contains("todo")) {
+                        isValidInput = true;
+                        createToDo(inputt, this.taskList);
+                    } else if (inputt.contains("deadline")) {
+                        isValidInput = true;
+                        createDeadline(inputt, this.taskList);
+                    } else if (inputt.contains("event")) {
+                        isValidInput = true;
+                        createEvent(inputt, this.taskList);
+                    }
+
+                    if (!isValidInput) {
+                        throw new IncorrectCommandException("I'm sorry, but I don't know what that means :-(");
+                    }
+                    //printing to terminal to notify user
```

Comment:
Java Coding Standard: Consider leaving a line to represent the chunk of code that performs `printing to terminal to notify user`